Android 13 deprecation warning

First of all thanks for all your hard work so far!  I updated to 0.5.1 via Flutter, in which I saw Android 12 support.  My phone is running Android 13 and this deprecation warning occurs:

~~~~
<user path>\AppData\Local\Pub\Cache\hosted\[pub.dev](http://pub.dev/)\usb_serial-0.5.1\android\src\main\java\dev\bessems\usbserial\UsbSerialPlugin.java:63: warning: [deprecation] <T>getParcelableExtra(String) in Intent has been deprecated
                        device = intent.getParcelableExtra(UsbManager.EXTRA_DEVICE);
<user path>\AppData\Local\Pub\Cache\hosted\[pub.dev](http://pub.dev/)\usb_serial-0.5.1\android\src\main\java\dev\bessems\usbserial\UsbSerialPlugin.java:80: warning: [deprecation] <T>getParcelableExtra(String) in Intent has been deprecated
                        device = intent.getParcelableExtra(UsbManager.EXTRA_DEVICE);
~~~~

The code is handled by the if statement, so I don't believe anything is actually broken, but it just creates noise in the build.  Everything works, and I have a PR to silence the warning.

--- a/android/src/main/java/dev/bessems/usbserial/UsbSerialPlugin.java
+++ b/android/src/main/java/dev/bessems/usbserial/UsbSerialPlugin.java
@@ -47,6 +47,17 @@ public class UsbSerialPlugin implements FlutterPlugin, MethodCallHandler, EventC
 
     private final BroadcastReceiver usbReceiver = new BroadcastReceiver() {
 
+        private UsbDevice getUsbDeviceFromIntent(Intent intent) {
+            if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.TIRAMISU) {
+                return intent.getParcelableExtra(UsbManager.EXTRA_DEVICE, UsbDevice.class);
+            } else {
+                // Create local variable to keep scope of deprecation suppression smallest
+                @SuppressWarnings("deprecation")
+                UsbDevice ret = intent.getParcelableExtra(UsbManager.EXTRA_DEVICE);
+                return ret;
+            }
+        }
+
         @Override
         public void onReceive(Context context, Intent intent) {
             String action = intent.getAction();
@@ -56,12 +67,7 @@ public class UsbSerialPlugin implements FlutterPlugin, MethodCallHandler, EventC
             if (action.equals(ACTION_USB_ATTACHED)) {
                 Log.d(TAG, "ACTION_USB_ATTACHED");
                 if (m_EventSink != null) {
-                    UsbDevice device;
-                    if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.TIRAMISU) {
-                        device = intent.getParcelableExtra(UsbManager.EXTRA_DEVICE, UsbDevice.class);
-                    } else {
-                        device = intent.getParcelableExtra(UsbManager.EXTRA_DEVICE);
-                    }
+                    UsbDevice device = getUsbDeviceFromIntent(intent);
                     if (device != null) {
                         HashMap<String, Object> msg = serializeDevice(device);
                         msg.put("event", ACTION_USB_ATTACHED);
@@ -73,12 +79,7 @@ public class UsbSerialPlugin implements FlutterPlugin, MethodCallHandler, EventC
             } else if (action.equals(ACTION_USB_DETACHED)) {
                 Log.d(TAG, "ACTION_USB_DETACHED");
                 if (m_EventSink != null) {
-                    UsbDevice device;
-                    if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.TIRAMISU) {
-                        device = intent.getParcelableExtra(UsbManager.EXTRA_DEVICE, UsbDevice.class);
-                    } else {
-                        device = intent.getParcelableExtra(UsbManager.EXTRA_DEVICE);
-                    }
+                    UsbDevice device = getUsbDeviceFromIntent(intent);
                     if (device != null) {
                         HashMap<String, Object> msg = serializeDevice(device);
                         msg.put("event", ACTION_USB_DETACHED);

